app/models/katello/activation_key.rb

```diff
@@ -99,11 +99,33 @@ def products
           Rails.logger.error("Pool #{pool.id} is missing its subscription id.")
         end
       end
-      all_products.flatten!
+      all_products.uniq.flatten
+    end
+
+    def all_products
+      Katello::Product.joins(:subscriptions => :pools).where(:organization_id => organization.id).enabled.uniq
     end
 
     def available_content
-      self.products ? self.products.map(&:available_content).flatten.uniq { |product| product.content.id } : []
+      self.products.map(&:available_content).flatten.uniq { |product| product.content.id }
+    end
+
+    def available_product_content(content_access_mode_all = false, content_access_mode_env = false)
```

Comment:
confused about these two methods 'available_content' and 'available_product_content', i see they do somewhat different things, but are they both needed?